test/e2e/specs/interactivity/router-regions.spec.ts

```diff
@@ -144,4 +198,125 @@ test.describe( 'Router regions', () => {
+			await expect( region ).toBeVisible();
+
+			const text = region.getByTestId( 'text' );
+			const clientCounter = region.getByTestId( 'client-counter' );
+			const serverCounter = region.getByTestId( 'server-counter' );
+
+			await expect( text ).toHaveText( regionId );
+			await expect( clientCounter ).toHaveText( '0' );
+			await expect( serverCounter ).toHaveText( '0' );
+
+			await clientCounter.click( { clickCount: 3, delay: 50 } );
+			await expect( clientCounter ).toHaveText( '3' );
+		}
+
+		// Region 6 has an init directive that should have been executed.
+		await expect( region6.getByTestId( 'has-directives' ) ).toBeVisible();
+
+		// Navigate to "Page attachTo 2".
+		await page.getByTestId( 'next' ).click();
+
+		// Check that regions remains hydrated and interactive.
+		for ( const regionId in regions ) {
+			const region = regions[ regionId ];
+			await expect( region ).toBeVisible();
+
+			const text = region.getByTestId( 'text' );
+			const clientCounter = region.getByTestId( 'client-counter' );
+			const serverCounter = region.getByTestId( 'server-counter' );
+
+			await expect( text ).toHaveText( regionId );
+			await expect( clientCounter ).toHaveText( '3' );
+			await expect( serverCounter ).toHaveText( '10' );
+
+			await clientCounter.click( { clickCount: 3, delay: 50 } );
+			await expect( clientCounter ).toHaveText( '6' );
+		}
+
+		// Navigate back to "Page attachTo 1".
+		await page.goBack();
+
+		// Check that regions remains hydrated and interactive.
+		for ( const regionId in regions ) {
+			const region = regions[ regionId ];
+			await expect( region ).toBeVisible();
+
+			const text = region.getByTestId( 'text' );
+			const clientCounter = region.getByTestId( 'client-counter' );
+			const serverCounter = region.getByTestId( 'server-counter' );
+
+			await expect( text ).toHaveText( regionId );
+			await expect( clientCounter ).toHaveText( '6' );
+			await expect( serverCounter ).toHaveText( '0' );
+
+			await clientCounter.click( { clickCount: 3, delay: 50 } );
+			await expect( clientCounter ).toHaveText( '9' );
+		}
+
+		// Navigate back to the initial page.
+		await page.goBack();
+
+		// Regions should be unmounted.
+		await expect( region3 ).toBeHidden();
+		await expect( region4 ).toBeHidden();
+		await expect( region5 ).toBeHidden();
+		await expect( region6 ).toBeHidden();
+
+		await page.goForward();
+
+		// Regions should b reset when mounted again.
+		for ( const regionId in regions ) {
+			const region = regions[ regionId ];
+			await expect( region ).toBeVisible();
+
+			const text = region.getByTestId( 'text' );
+			const clientCounter = region.getByTestId( 'client-counter' );
+			const serverCounter = region.getByTestId( 'server-counter' );
+
+			await expect( text ).toHaveText( regionId );
+			await expect( clientCounter ).toHaveText( '0' );
+			await expect( serverCounter ).toHaveText( '0' );
+		}
+
+		// Region 6 has an init directive that should have been executed.
```

Comment:
Maybe this check should be moved outside of the regions, so we can check that it didn't happen on the previous navigation? It could use `state` instead of `context` and add 1 to a counter instead of `true`.

Reply:
Done at https://github.com/woocommerce/gutenberg/pull/14/commits/e2b7bb80f83cbc6237e51472716d0957683799bc.

Reply:
Oh, what I meant was to add a `await expect( initCount ).toHaveText( '1' );` in [after this navigation](https://github.com/woocommerce/gutenberg/blob/e2b7bb80f83cbc6237e51472716d0957683799bc/test/e2e/specs/interactivity/router-regions.spec.ts#L306-L315) to make sure that it was not (re)executed. 

Never mind, it's not critical 🙂

Reply:
Oops! 😅 Well, there's another check after going back to the initial page and forward again. The counter should be '2' at that moment, and it passes. I guess that's safe enough. 🤞

Reply:
Oh, that's right 🙂